File: DotNet/CesiumLanguageWriter/Advanced/CesiumHeightReferenceValuePropertyAdaptor.cs
using JetBrains.Annotations;

namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// Adapts a class that implements <see cref="ICesiumHeightReferenceValuePropertyWriter"/> to implement
    /// <see cref="ICesiumPropertyWriter"/> for <see cref="CesiumHeightReference"/> values.
    /// </summary>
    /// <typeparam name="TFrom">The class that implements <see cref="ICesiumHeightReferenceValuePropertyWriter"/> to adapt.</typeparam>
    public class CesiumHeightReferenceValuePropertyAdaptor<TFrom> : CesiumWriterAdaptor<TFrom, CesiumHeightReference>
        where TFrom : class, ICesiumHeightReferenceValuePropertyWriter
    {
        /// <summary>
        /// Initializes a new instance.
        /// </summary>
        /// <param name="parent">The instance to wrap.</param>
        /// <param name="writeValueCallback">The callback to write values of type <see cref="CesiumArcType"/>.</param>
        /// <param name="writeDeleteValueCallback">The callback to write an indication that the client should delete existing data.</param>
        public CesiumHeightReferenceValuePropertyAdaptor([NotNull] TFrom parent,
                                                         [NotNull] CesiumWriterAdaptorWriteCallback<TFrom, CesiumHeightReference> writeValueCallback,
                                                         [NotNull] CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback)
            : base(parent, writeValueCallback, writeDeleteValueCallback)
        {
        }
    }
}

File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/CesiumHeightReferenceValuePropertyAdaptor.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import javax.annotation.Nonnull;

/**
 *  
 Adapts a class that implements {@link ICesiumHeightReferenceValuePropertyWriter} to implement
 {@link ICesiumPropertyWriter} for {@link CesiumHeightReference} values.
 
 

 * @param <TFrom> The class that implements {@link ICesiumHeightReferenceValuePropertyWriter} to adapt.
 */
@SuppressWarnings( {
        "unused",
        "deprecation",
        "serial"
})
public class CesiumHeightReferenceValuePropertyAdaptor<TFrom extends ICesiumHeightReferenceValuePropertyWriter> extends CesiumWriterAdaptor<TFrom, CesiumHeightReference> {
    /**
    *  
    Initializes a new instance.
    
    
    
    

    * @param parent The instance to wrap.
    * @param writeValueCallback The callback to write values of type {@link CesiumHeightReference}.
    * @param writeDeleteValueCallback The callback to write an indication that the client should delete existing data.
    */
    public CesiumHeightReferenceValuePropertyAdaptor(@Nonnull TFrom parent, @Nonnull CesiumWriterAdaptorWriteCallback<TFrom, CesiumHeightReference> writeValueCallback,
            @Nonnull CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback) {
        super(parent, writeValueCallback, writeDeleteValueCallback);
    }
}